lib/dns.js

```diff
@@ -311,26 +318,31 @@ function setServers(servers) {
   const orig = cares.getServers();
   const newSet = [];
   const IPv6RE = /\[(.*)\]/;
-  const addrSplitRE = /:\d+$/;
+  const addrSplitRE = /(^.+?)(?::(\d+))?$/;
```

Comment:
Not sure I follow? Just ignore any additional input. The host and the port are the only bits we care about. If you do not think it is appropriate to ignore the rest, then throw if values for the other parts are provided.